flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/validation/DefaultValidator.java

```diff
@@ -453,18 +446,6 @@ private Optional<String> validateSpecChange(
         JobSpec oldJob = oldSpec.getJob();
         JobSpec newJob = newSpec.getJob();
         if (oldJob != null && newJob != null) {
-            if (StringUtils.isNullOrWhitespaceOnly(
-                            effectiveConfig.get(CheckpointingOptions.SAVEPOINT_DIRECTORY.key()))
-                    && deployment.getStatus().getJobManagerDeploymentStatus()
-                            != JobManagerDeploymentStatus.MISSING
-                    && ReconciliationUtils.isUpgradeModeChangedToLastStateAndHADisabledPreviously(
```

Comment:
nit: `ReconciliationUtils.isUpgradeModeChangedToLastStateAndHADisabledPreviously` can be safely removed